commercial/app/model/commercial/books/BookFinder.scala

```diff
@@ -2,29 +2,29 @@ package model.commercial.books
 
 import common.{AkkaAgent, ExecutionContexts, Logging}
 import conf.Configuration
-import play.api.Play.current
+import conf.Switches.GuBookshopFeedsSwitch
+import model.commercial.{FeedReader, FeedRequest}
 import play.api.libs.json._
 import play.api.libs.oauth.{ConsumerKey, OAuthCalculator, RequestToken}
-import play.api.libs.ws.WS
 
 import scala.concurrent.Future
+import scala.concurrent.duration._
 
 object BookFinder extends ExecutionContexts {
 
   private lazy val agent = AkkaAgent[Map[String, Book]](Map.empty[String, Book])
 
-  def findByIsbn(isbn: String): Future[Option[Book]] = {
-    val book = agent.get().get(isbn)
-    if (book.isDefined) {
-      Future.successful(book)
-    } else {
-      MagentoService.findByIsbn(isbn) map {
-        _ map { foundBook =>
-          agent.send { books => books + (isbn -> foundBook)}
-          foundBook
-        }
-      }
+  def findByIsbn(isbn: String): Option[Book] = {
+    val maybeBook = agent.get().get(isbn)
+
+    if (maybeBook.isEmpty) {
```

Comment:
This will still "dogpile" if the book service slows down. maybe we should have one of those circuit breakers ala @janua & @robertberry